2.0/s2i/bin/run

```diff
@@ -1,5 +1,26 @@
 #!/bin/bash
 
+function export_memory_limit() {
+  if [ -f /sys/fs/cgroup/memory/memory.limit_in_bytes ]; then
+    export MEMORY_LIMIT_IN_BYTES=$(cat /sys/fs/cgroup/memory/memory.limit_in_bytes)
+    return 0
+  fi
+  return 1
+}
+
+function export_number_of_cores() {
+  if [ -f /sys/fs/cgroup/cpuset/cpuset.cpus ]; then
+    export NUMBER_OF_CORES=`awk ' \
+      BEGIN { i = 0; FS="-"; RS="[,\n]" } \
+      /^[0-9]+-[0-9]+$/ { i += $2-$1+1 }  \
+      /^[0-9]+$/        { ++i }           \
+      END { print i} \
+      ' /sys/fs/cgroup/cpuset/cpuset.cpus`
+    return 0
+  fi
+  return 1
+}
```

Comment:
I think i'd expect them to just return values instead of setting env variables, but otherwise they look ok.

(I guess if they returned values i'm not sure how you'd handle error conditions though, so maybe it's best as is)